core/src/main/java/org/bf2/srs/fleetmanager/execution/impl/workers/deprovision/StartDeprovisionRegistryWorker.java

```diff
@@ -55,31 +60,46 @@ public void execute(Task aTask, WorkerContext ctl) throws RegistryStorageConflic
         if (registryOptional.isPresent()) { // FAILURE POINT 1
 
             var registry = registryOptional.get();
+
+            var force = registry.getCreatedAt()
+                    .plus(props.getDeprovisionStuckInstanceTimeout())
+                    .isAfter(Instant.now());
+            if (force) {
+                log.warn("Registry instance '{}' is forced to be deprovisioned.", registry);
```

Comment:
This means, that multiple deprovisioning tasks could theoretically be executed at the same time, after the timeout expires. I'll need to figure out if calling:
`org.bf2.srs.fleetmanager.spi.TenantManagerService#deleteTenant` and
`org.bf2.srs.fleetmanager.spi.AccountManagementService#deleteSubscription` is reentrant.
It seems that it should be. Any opinions?

Reply:
what do you mean by reentrant? do you mean that can be called multiple times but the result will always be the same? if it's that I think yes, or at least it shouldn't cause any side effects or any harm

Reply:
> do you mean that can be called multiple times but the result will always be the same?
Exactly, since the task will no longer block when deprovisioning is already in progress

Reply:
You can call deleteSubscription as many times as you want since we're doing some sort of fire and forget there. In the worst-case scenario, the subscription has been deleted already and nothing happens.